### DISCUSSION TITLE
'npx prisma generate' command failed

### DISCUSSION THREAD

**Discussion by moezbenrebah** [prisma/prisma#19755] [Q&A]
### Bug description

I have a react app that use Prisma in order to search for GitHub repositories, when I run 'npm run start' that will start a server on the backend and the react app in the frontend, I've got this error:
```
[start:client] Proxy error: Could not proxy request /api/repositories?q=stars:%3E10000+language:rust&per_page=10 from localhost:3000 to http://localhost:8000.
```
So, after trying to debug this error: I found that I have to run the 'npx prisma generate' but it fails with the error:
```
> Downloading Prisma engines for Node-API for dError: request to https://binaries.prisma.sh/all_commits/39190b250ebc338586e25e6da45e5e783bc8a635/debian-openssl-3.0.x/migration-engine.gz.sha256 failed, reason: connect ETIMEDOUT
```
I almost trying everything to find a solution but I'm still blocked.

### How to reproduce

npx prisma generate

### Prisma information

this is my package.json file:
```
{
  "name": "codesplain",
  "version": "0.1.0",
  "private": true,
  "proxy": "http://localhost:8000",
  "dependencies": {
    "@exuanbo/file-icons-js": "^3.3.0",
    "@monaco-editor/react": "^4.4.6",
    "@playwright/test": "^1.28.1",
    "@primer/octicons-react": "^17.9.0",
    "@prisma/client": "^4.7.0",
    "@tailwindcss/forms": "^0.5.3",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "axios": "0.21.4",
    "classnames": "^2.3.2",
    "concurrently": "^7.6.0",
    "cookie-session": "^2.0.0",
    "dotenv": "^16.0.3",
    "express": "^4.18.2",
    "msw": "^0.49.2",
    "nodemon": "^2.0.20",
    "playwright": "^1.28.1",
    "prisma": "^4.7.0",
    "prop-types": "^15.8.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router": "^6.4.4",
    "react-router-dom": "^6.4.4",
    "react-scripts": "^5.0.1",
    "react-split": "^2.0.14",
    "swr": "^2.0.0",
    "validate.js": "^0.13.1",
    "web-vitals": "^2.1.4"
  },
  "prisma": {
    "schema": "server/prisma/schema.prisma"
  },
  "scripts": {
    "start": "concurrently \"npm:start:server\" \"npm:start:client\"",
    "start:client": "react-scripts start",
    "start:server": "nodemon --watch server server/index.mjs",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "autoprefixer": "^10.4.13",
    "postcss": "^8.4.19",
    "tailwindcss": "^3.2.4"
  }
}
```

### Environment & setup

I'm under Ubuntu 22.04

### Prisma Version

```
prisma version 4.7.0
```

**Comment by janpio** [comment]
Can you manually request that URL and download the file?

**Comment by moezbenrebah** [comment]
wget http://localhost:8000/api/repositories?q=react

--2023-06-14 12:49:27--  http://localhost:8000/api/repositories?q=react
Resolving localhost (localhost)... 127.0.0.1
Connecting to localhost (localhost)|127.0.0.1|:8000... failed: Connection refused.

**Reply by janpio** [reply]
I meant the URL that can not be accessed during `prisma generate`: https://binaries.prisma.sh/all_commits/39190b250ebc338586e25e6da45e5e783bc8a635/debian-openssl-3.0.x/migration-engine.gz.sha256

**Reply by moezbenrebah** [reply]
it download a file 'migration-engine.gz.sha256' with a content:
"154c89a19b64703a4e66532166291939df5fb4bd2b6ab2c96e89564f7d8f2a64  migration-engine.gz"

**Reply by moezbenrebah** [reply]
what should I do then?

**Reply by janpio** [reply]
This highlights that your Prisma CLI running on your command line for some reason can not connect to that URL in the configured timeout, but that nothing is wrong with the URL itself.

**Reply by moezbenrebah** [reply]
do you think the solution suggested in this issue: https://github.com/reworkd/AgentGPT/issues/100 can fix my error, and please check the versions of the dependencies because I use prisma 4.7.0

**Reply by janpio** [reply]
That would be a valid workaround, to download the files manually and put them into a local spot and then use the environment variables to point to these local files. That will skip the download.

Fundamentally this should not be necessary though, and with a future Prisma version you will have to repeat this process every time. So better figure out why it is timing out. Can you maybe try on another, faster, internet connection? Or just on another computer using the same internet connection?

**Reply by moezbenrebah** [reply]
my internet connect speed is:
90.5
Download in Mbit/s

78.9
Transfert in Mbit/s
How about to switch to Windows OS?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
